I/IGLWrap/build_tarballs.jl

```diff
@@ -0,0 +1,62 @@
+#=
+BINARYBUILDER_RUNNER=privileged julia build_tarballs.jl --verbose --debug       
+=#
+
+using BinaryBuilder
+using Pkg
+
+version = v"2.3.0" # libIGL version, see below:
+
+sources = [
+	DirectorySource("./iglwrap"),
+	# 2.3.0 stable release:
+	GitSource("https://github.com/libigl/libigl.git",
+		"e60423e28c86b6aa2a3f6eb0112e8fd881f96777")
+]
+
+script = raw"""
+cd $WORKSPACE/srcdir
+mkdir -p build
+cd build
+# this line seems to prevent compilation; removed for testing:
+# -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
+cmake \
+    -DCMAKE_INSTALL_PREFIX=${prefix} \
+    -DCMAKE_MODULE_PATH=${WORKSPACE}/srcdir/libigl/cmake\; \
+    ..
+make -j${nproc}
+make install
+
+# install_license ${WORKSPACE}/srcdir/LICENSE
+"""
+
+products = [
+	LibraryProduct("libiglwrap", :libiglwrap;
+		dlopen_flags=[:RTLD_NOW,:RTLD_LOCAL])
+]
+
+platforms = [
+	# 2021-10-30: LibIGL won't compile on at least MacOS right now, but
+	# works on the following platforms:
+	Platform("x86_64", "linux"),
+]
+# platforms = supported_platforms()
+
+platforms = expand_cxxstring_abis(platforms)
+
+
+dependencies = [
+  Dependency("boost_jll", compat="~1.71"),
+	Dependency("GMP_jll"),
+	Dependency("MPFR_jll"),
+	Dependency("OpenBLAS_jll"),
+	Dependency("CGAL_jll"),
```

Comment:
Alright, CGAL doesn't seem to be actually used at runtime:
```console
% readelf -a libiglwrap.so | grep NEEDED 
 0x0000000000000001 (NEEDED)             Shared library: [libboost_thread.so.1.71.0]
 0x0000000000000001 (NEEDED)             Shared library: [libpthread.so.0]
 0x0000000000000001 (NEEDED)             Shared library: [libgmp.so.10]
 0x0000000000000001 (NEEDED)             Shared library: [libmpfr.so.6]
 0x0000000000000001 (NEEDED)             Shared library: [libstdc++.so.6]
 0x0000000000000001 (NEEDED)             Shared library: [libm.so.6]
 0x0000000000000001 (NEEDED)             Shared library: [libgcc_s.so.1]
 0x0000000000000001 (NEEDED)             Shared library: [libc.so.6]
 0x0000000000000001 (NEEDED)             Shared library: [ld-linux-x86-64.so.2]
```
And the log says
```
[18:11:57] -- Using header-only CGAL
```
which seems to confirm that.  If this is the case, then we don't need CGAL as a runtime dependency and it's ok(-ish) to have them downloading a random version of CGAL
```suggestion
```
If this wasn't the case, we would end up loading two different CGAL versions, which is begging for troubles (hence why I asked to add CGAL before).

Reply:
I can confirm that everything is used as header-only dependencies (the C++ code is extremely heavily templated). I tried looking in IGL documentation/issues about how to provide an already-existing CGAL instead of downloading it, and only found that they did lots of work in the exact opposite direction (their goal being to make the cmake process as simple as possible), e.g. https://github.com/libigl/libigl/issues/563

(by the way, the `CGAL_jll` package is *also* header-only: it declares a single artifact, with the name `CGAL.v5.3.0.any.tar.gz`; so no interference was possible anyway).

Reply:
> by the way, the CGAL_jll package is also header-only

Right, that's a recent change: #3284